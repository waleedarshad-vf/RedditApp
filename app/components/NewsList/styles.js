import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  list: {
    flex: 1,
    padding: 10,
    justifyContent: "flex-start",
    width: "100%",
    backgroundColor: "#cee3f8"
  }
});

export default styles;
