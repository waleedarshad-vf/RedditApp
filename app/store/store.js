import { applyMiddleware, combineReducers, createStore } from "redux";
import thunk from "redux-thunk";
import logger from "redux-logger";
import { newsReducer } from "./reducer/NewsReducer.js";

let rootReducer = combineReducers({
  news: newsReducer
});
const store = createStore(rootReducer, applyMiddleware(logger, thunk));

export default store;
