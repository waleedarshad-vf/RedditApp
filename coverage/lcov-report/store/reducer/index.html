<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for store/reducer</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> store/reducer
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">39.39% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>13/33</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">27.78% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>5/18</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">16.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">87.5% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>7/8</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="NewsReducer.js"><a href="NewsReducer.js.html">NewsReducer.js</a></td>
	<td data-value="39.39" class="pic low"><div class="chart"><div class="cover-fill" style="width: 39%;"></div><div class="cover-empty" style="width:61%;"></div></div></td>
	<td data-value="39.39" class="pct low">39.39%</td>
	<td data-value="33" class="abs low">13/33</td>
	<td data-value="27.78" class="pct low">27.78%</td>
	<td data-value="18" class="abs low">5/18</td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="6" class="abs low">1/6</td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="8" class="abs high">7/8</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Jul 22 2019 16:13:33 GMT+0500 (Pakistan Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
