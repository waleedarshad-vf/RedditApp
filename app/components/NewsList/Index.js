import React, { Component } from "react";
import { View, Text, FlatList } from "react-native";
import { connect } from "react-redux";
import styles from "./styles";
import NewsItem from "../NewsItem";
import { getList } from "../../store/actions/NewsAction";
class NewsList extends Component {
  state = { selected: (new Map(): Map<string, boolean>) };

  _keyExtractor = (item, index) => item.id;

  _onPressItem = (id: string) => {
    this.setState(state => {
      const selected = new Map(state.selected);
      selected.set(id, !selected.get(id));
      return { selected };
    });
  };

  _renderItem = ({ item }) => (
    <NewsItem
      id={item.id}
      onPressItem={this._onPressItem}
      selected={!!this.state.selected.get(item.id)}
      title={item.title}
    />
  );
  componentWillMount() {
    this.props.getList();
  }

  componentWillReceiveProps(nextProps) {
    console.log("nextProps", nextProps);
  }
  render() {
    return (
      <FlatList
        data={this.props.data}
        extraData={this.state}
        keyExtractor={this._keyExtractor}
        renderItem={this._renderItem}
      />
    );
  }
}

const mapStateToProps = state => {
  const { news } = state;
  return { news };
};

export default connect(
  mapStateToProps,
  { getList }
)(NewsList);
