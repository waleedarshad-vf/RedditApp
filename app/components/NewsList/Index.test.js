import React from "react";
import { shallow } from "enzyme";
import NewsList from "./Index";
import { Provider } from "react-redux";
import store from "../../store/store.js";
describe("NewsList", () => {
  describe("Rendering", () => {
    it("should match to snapshot", () => {
      const component = shallow(
        <Provider store={store}>
          <NewsList />
        </Provider>
      );
      expect(component).toMatchSnapshot();
    });
  });
});
