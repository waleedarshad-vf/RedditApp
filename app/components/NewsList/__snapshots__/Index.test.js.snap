// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsList Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
